pkg/hostagent/dns/dns_test.go

```diff
@@ -0,0 +1,115 @@
+package dns
+
+import (
+	"io"
+	"log"
+	"net"
+	"regexp"
+	"testing"
+
+	"github.com/foxcpp/go-mockdns"
+	"github.com/miekg/dns"
+	"github.com/stretchr/testify/assert"
```

Comment:
gotest.tools/v3/assert

Reply:
I was originally trying to use `gotest.tools/v3/assert` originally but had some difficulty getting an assertion that worked and testify had a straight-forward option to match the regex to the dns.Msg object (interface?).  Taking another pass at it now, I was able to get a successful way to test with `gotest.tools/v3/assert` :)